packages/astro/src/@types/astro.ts

```diff
@@ -1375,6 +1375,72 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		devOverlay?: boolean;
+
+		// TODO review with docs team before merging to `main`
+		/**
+		 * @docs
+		 * @name experimental.i18n
+		 * @type {object}
```

Comment:
Exactly, no plain boolean. We talked about it internally, and we're happy to ship the configuration like this because it's possible we are going to change the names of the configurations too.